python-package/lightgbm/basic.py

```diff
@@ -2585,6 +2585,12 @@ def _dump_text(self, filename: Union[str, Path]) -> "Dataset":
         return self
 
 
+_LGBM_CustomObjectiveFunction = Callable[
```

Comment:
can you please remove this and the changes to `update()` and `__boost()`? Those changes are from #5359, which has already been merged to `master`, so I'm not sure why they're showing up in the diff.

The easiest way to remove them is probably to merge `master` into your branch.

```shell
git checkout feature/static-typing
git remote add upstream https://github.com/microsoft/LightGBM.git
git pull upstream master
git push origin feature/static-typing
```

Reply:
It's done!

Reply:
I don't believe so. These lines are still showing up in the diff.

for example:

<img width="570" alt="image" src="https://user-images.githubusercontent.com/7608904/179368367-91bf10cf-1a73-45e6-abd9-10cf4c577dc6.png">

You may need to do something like the following.

```shell
git rebase master
git push origin feature/static-typing --force
```

Reply:
Something weird happened the first time, but I think it's already fixed
<img width="816" alt="Screen Shot 2022-07-16 at 17 38 42" src="https://user-images.githubusercontent.com/1332029/179371260-89352a0f-a12b-429f-b0d6-505bfbac1193.png">
.

Reply:
If you navigate to https://github.com/microsoft/LightGBM/pull/5362/files in your browser, you will see that chaanges too `__boost()` and `update()` are still present in the diff.

You can see that in the statistics as well.

<img width="113" alt="image" src="https://user-images.githubusercontent.com/7608904/179371362-694c79e5-fce3-4c73-8f3b-a7d7a9a5aa20.png">

I'd expect this to be `+6, -6` since you've made 6 one-line changes.

If rebasing isn't working or you aren't sure how to do it, I'd be happy to help! If you go to https://github.com/makquel/LightGBM/settings/access and grant me write access on your fork, I can do this git work and and push changes to your branch.

Reply:
I made it manually, so I believe it's finally fixed!

Reply:
looks great now, thanks very much for working through that!